source/SonicHeroes.WidescreenFix/dllmain.cpp

```diff
@@ -1404,6 +1406,17 @@ void Init()
 		injector::WriteMemory<uint32_t>(ptr_722FA0, 0, true);
 	}
 
+	if (WarpSwitchEffectMode)
+	{
+		uintptr_t loc_41BAD5 = reinterpret_cast<uintptr_t>(hook::pattern("83 7F 50 04 75 0D 6A 02 6A 0A E8 ? ? ? ? 6A 06 EB 0B 6A 05 6A 0A E8 ? ? ? ? 6A 02 6A 0B").get_first(0)) + 0x10;
+		int8_t EffectMode = 2;
+
+		if (WarpSwitchEffectMode >=2)
+			EffectMode = 1;
```

Comment:
This is unnecessary, just let the inireader write into `WarpSwitchEffectMode` and use it directly.